source/runtime.cpp

```diff
@@ -973,7 +970,7 @@ void reshade::runtime::load_preset(const std::filesystem::path &path)
 	if (_reload_remaining_effects != 0 && // ... unless this is the 'load_current_preset' call in 'update_and_render_effects'
 		(_performance_mode || preset_preprocessor_definitions != _preset_preprocessor_definitions))
 	{
-		assert(path == _current_preset_path);
+		assert(preset.path() == _current_preset_path);
```

Comment:
This assert is no longer necessary, since the path is guaranteed to always be `_current_preset_path` (since it's opened with that above). So can get rid of it.
Conveniently it's also the only use of `ini_file::path()`, so can get rid of that function too.